### PR TITLE
Issue #3316: [UX] Add human name to fields list at admin/reports/fields

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/3316

### CODE REVIEW

File: core/modules/field_ui/field_ui.admin.inc [hunk 1]
```diff
@@ -45,26 +46,39 @@ function field_ui_fields_list() {
...
+          $rows[$field_name]['data'][1] = $field_types[$field['type']]['label'];
+          $rows[$field_name]['data'][2] = $modules[$module_name]->info['name'];
         }
 
         // Add the current instance.
         $admin_path = _field_ui_bundle_admin_path($entity_type, $bundle);
-        $rows[$field_name]['data'][2][] = $admin_path ? l($bundles[$entity_type][$bundle]['label'], $admin_path . '/fields') : $bundles[$entity_type][$bundle]['label'];
+        $field_label = $instances[$entity_type][$bundle][$field_name]['label'];
+        $bundle_label = $bundles[$entity_type][$bundle]['label'];
+        if ($admin_path) {
```
Comment (klonos) [inline core/modules/field_ui/field_ui.admin.inc:56]: This is to account for the previous `$rows[$field_name]['data'][2][] = $admin_path ? ...` conditional that would render the bundle name/label as a link to the "Manage fields" page of that bundle, or render it as plain text.

Not sure which cases that was there to cover. Was it permissions (current used not allowed to edit the bundle)? ...or are there cases of bundles/fields (perhaps in contrib if not in core?) that cannot be configured? Anyway, I though that there must have been a reason, so I have retained it. If there is no reason, then we could remove the conditional, and have the field label and bundle label be rendered as links in all cases.

File: core/modules/field_ui/field_ui.admin.inc [hunk 2]
```diff
@@ -45,26 +46,39 @@ function field_ui_fields_list() {
...
+        else {
+          $rows[$field_name]['data'][3][] = t('%field_label in @bundle_label', array('%field_label' => $field_label, '@bundle_label' => $bundle_label));
+        }
       }
     }
   }
   foreach ($rows as $field_name => $cell) {
-    $rows[$field_name]['data'][2] = implode(', ', $cell['data'][2]);
+    // Render the items of the "Used in" column as a list.
+    $rows[$field_name]['data'][3] = theme('item_list', array('items' => $rows[$field_name]['data'][3]));
```
Comment (klonos) [inline core/modules/field_ui/field_ui.admin.inc]: As explained in the issue, a reused filed can have different label across the various bundles it is being used in.

**Comment by klonos** [comment]
Time for a sandbox refresh 😄

File: core/modules/field_ui/field_ui.admin.inc [hunk 3]
```diff
@@ -57,28 +56,22 @@ function field_ui_fields_list() {
...
 
-  if (empty($rows)) {
-    $output = t('No fields have been defined yet.');
-  }
-  else {
-    // Sort rows by field name.
-    ksort($rows);
-    $output = theme('table', array('header' => $header, 'rows' => $rows, 'empty' => t('No fields available.'),));
-  }
+  $output = theme('table', array('header' => $header, 'rows' => $rows, 'empty' => t('No fields have been defined yet.'),));
```
Comment (opi) [inline core/modules/field_ui/field_ui.admin.inc:74]: get sort+order with `tablesort_get_sort` & `tablesort_get_order` and the properly sort the `$rows` array before rendering as table.

**Comment by klonos** [comment]
Closing in favor of https://github.com/backdrop/backdrop/pull/2528

**Pull request closed without merging**